haystack/tools/from_function.py

```diff
@@ -210,9 +210,16 @@ def _remove_title_from_schema(schema: Dict[str, Any]):
     :param schema:
         The JSON schema to remove the 'title' keyword from.
     """
-    schema.pop("title", None)
-
-    for property_schema in schema["properties"].values():
-        for key in list(property_schema.keys()):
-            if key == "title":
-                del property_schema[key]
+    for key, value in list(schema.items()):
+        # Make sure not to remove parameters named title
+        if key == "properties" and isinstance(value, dict) and "title" in value:
+            for sub_val in value.values():
+                _remove_title_from_schema(sub_val)
```

Comment:
Why don't we also remove title from schema recursively as post-processing hook rather than having so many explicit calls?

Reply:
Ah sorry never mind, this is the recursive function 🤦